lms/djangoapps/discussion_api/api.py

```diff
@@ -41,17 +42,18 @@
 from lms.lib.comment_client.thread import Thread
 from lms.lib.comment_client.utils import CommentClientRequestError
 from openedx.core.djangoapps.course_groups.cohorts import get_cohort_id
+from openedx.core.lib.exceptions import CourseNotFoundError, PageNotFoundError
 
 
-def _get_course_or_404(course_key, user):
+def _get_course(course_key, user):
     """
-    Get the course descriptor, raising Http404 if the course is not found,
-    the user cannot access forums for the course, or the discussion tab is
-    disabled for the course.
+    Get the course descriptor, raising CourseNotFoundError if the course is not found or
+    the user cannot access forums for the course, and DiscussionDisabledError if the
+    discussion tab is disabled for the course.
     """
     course = get_course_with_access(user, 'load', course_key, check_if_enrolled=True)
```

Comment:
Shouldn't you wrap this call within a try-except clause to trap the `Http404` error and replace it with a `CourseNotFoundError`?   That is what the docstring says.  Also, by doing so within this method, you don't need to have a try-except around all callers of this method.